lib/token.cpp

```diff
@@ -2283,6 +2283,11 @@ std::pair<const Token*, const Token*> Token::typeDecl(const Token * tok)
                 if (vt && vt->smartPointerTypeToken)
                     return { vt->smartPointerTypeToken, vt->smartPointerTypeToken->linkAt(-1) };
             }
+            if (astIsIterator(var->nameToken())) {
+              const ValueType* vt = var->valueType();
+              if (vt && vt->containerTypeToken)
+                  return { vt->containerTypeToken, vt->containerTypeToken->linkAt(-1) };
```

Comment:
This is not the correct type though. This will make the iterator type be the same as the inner type which is not correct.